middleware/helpers.js

```diff
@@ -50,3 +50,35 @@ module.exports.getMaxDate = () => {
 
   return `${year}-${month}-${day}`;
 };
+
+function checkMediaLink(mediaLink, linkRegex) {
+  const match = mediaLink.match(linkRegex);
+  return match && match[3];
+}
+
+module.exports.mediaLinkValidation = (facebook, twitter, instagram, tumblr) => {
+  const mediaRegex = {
+    facebook:
+      /^(https?:\/\/)?(www\.)?facebook\.com\/(?:[a-zA-Z0-9.]+\/)?(?:profile\.php\?id=(\d+)|([a-zA-Z0-9.]+))\/?$/,
```

Comment:
Bear in mind that media links can come with query parameters

Reply:
Since these are supposed to be links that lead specifically to the user account, shouldn't those links be absent of any sort of query params?

Reply:
I'm not 100% sure but it's pretty common to see media links with referral information attached to the url with query parameters. A tech savvy person would probably notice when they copy the URL but it's not fool proof for most. The pattern might fail if there are query params.

Reply:
I guess what could be done is the regex could be modified to look for potential query params and edit the provided link with those filtered out.